src/devices/cpu/pic16x8x/pic16x8x.cpp

```diff
@@ -0,0 +1,1443 @@
+		memcpy(m_buff, m_region->base() + eeprom_dump, m_internal_eeprom_size * 2);
+
+		// Data width conversion
+		for (u8 i=0; i < m_internal_eeprom_size; i++)
+			m_eeprom_data[i] = m_buff[i] & 0x00ff;
+	}
+
+}
+
+bool pic16x8x_device::nvram_read(util::read_stream &file)
+{
+	auto const [err, actual] = read(file, m_eeprom_data, m_internal_eeprom_size);
+	return !err && (actual == m_internal_eeprom_size);
+}
+
+bool pic16x8x_device::nvram_write(util::write_stream &file)
+{
+	auto const [err, actual] = write(file, m_eeprom_data, m_internal_eeprom_size);
+	return !err && (actual == m_internal_eeprom_size);
+}
+
+
+// EEPROM data access
+u8 pic16x8x_device::m_eeread(offs_t offs)
+{
+	return m_eeprom_data[offs];
+}
+
+void pic16x8x_device::m_eewrite(offs_t offs, u8 data)
+{
+	m_eeprom_data[offs] = data & m_data_mask;
+}
+
+
+// ******** The following is the STATUS flag register definition. ********
+// |  7  |  6  |  5  | 4 | 3 | 2 |  1 | 0 |
+// | IRP | RP1 | RP0 | T0| PD| Z | DC | C |
+
+// IRP Register Bank Select - Unimplemented on this device
+#define RP1_FLAG    0x40    // Register Bank Select Bit 1
+#define RP0_FLAG    0x20    // Register Bank Select Bit 0
+							// 00 = Bank 0 (00h-7fh)
+							// 01 = Bank 1 (80h-ffh)
+							// 10 = Bank 2 (100h-17fh)
+							// 11 = Bank 3 (180h-1ffh)
+							// Each bank is 128 bytes. Only bit RP0 is used by the PIC16F8X. RP1 should be mantained clear.
+#define TO_FLAG     0x10    // TO   Time Out flag (WatchDog)
+#define PD_FLAG     0x08    // PD   Power Down flag
+#define Z_FLAG      0x04    // Z    Zero Flag
+#define DC_FLAG     0x02    // DC   Digit Carry/Borrow flag (Nibble)
+#define C_FLAG      0x01    // C    Carry/Borrow Flag (Byte)
+
+#define RP0     (m_STATUS & RP0_FLAG)
+#define TO      (m_STATUS & TO_FLAG)
+#define PD      (m_STATUS & PD_FLAG)
+#define ZERO    (m_STATUS & Z_FLAG)
+#define DC      (m_STATUS & DC_FLAG)
+#define CARRY   (m_STATUS & C_FLAG)
+
+
+// ******** The following is the OPTION flag register definition. ********
+// |   7  |    6   |   5  |   4  |  3  |   2 |   1 |   0 |
+// | RBPU | INTEDG | T0CS | T0SE | PSA | PS2 | PS1 | PS0 |
+#define RBPU_FLAG    0x80   // RBPU PORTB Pull-up Enable bit
+#define INTEDG_FLAG  0x40   // INTEDG  Interrupt Edge Select bit
+							// 1 = Interrupt on rising edge of RB0/INT pin
+							// 0 = Interrupt on falling edge of RB0/INT pin
+#define T0CS_FLAG    0x20   // T0CS Timer 0 clock source select
+							// 1 = Transition on RA4/T0CKI pin
+							// 0 = Internal instruction cycle clock (CLKOUT)
+#define T0SE_FLAG    0x10   // T0SE Timer 0 clock source edge select
+							// 1 = Increment on high-to-low transition on RA4/T0CKI pin
+							// 0 = Increment on low-to-high transition on RA4/T0CKI pin
+#define PSA_FLAG     0x08   // PSA Prescaler Assignment bit
+							// 1 = Prescaler is assigned to the WDT
+							// 0 = Prescaler is assigned to the Timer0 module
+#define PS_REG       0x07   // PS Prescaler Rate select

```

Comment:
It's generally recommended to use `constexpr` instead of `#define` for constants in new code.

Reply:
working on this

Reply:
Done.